packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,326 @@
+/// tap and collision detection. Thus, the [size] should be set equal to
+/// the approximate bounding rectangle of the rendered picture. If you
+/// do not specify the size of a PositionComponent, then it will be
+/// equal to zero and the component won't be able to respond to taps.
+///
+class PositionComponent extends BaseComponent {
+  final Transform2D _transform;
+
+  PositionComponent({
+    Vector2? position,
+    Vector2? size,
+    Vector2? scale,
+    double angle = 0.0,
+    Anchor anchor = Anchor.topLeft,
+    int? priority,
+  })  : _transform = Transform2D(),
+        _anchor = anchor,
+        _size = NotifyingVector2.copy(size ?? Vector2.zero()),
+        _scaledSize = Vector2.zero(),
+        super(priority: priority) {
+    if (position != null) {
+      _transform.position = position;
     }
-    return _scaledSizeCache.value!;
+    if (angle != 0) {
+      _transform.angle = angle;
+    }
+    if (scale != null) {
+      _transform.scale = scale;
+    }
+    _size.addListener(_onModifiedSizeOrAnchor);
+    _size.addListener(_onModifiedSizeOrScale);
+    _transform.scale.addListener(_onModifiedSizeOrScale);
+    _onModifiedSizeOrAnchor();
+    _onModifiedSizeOrScale();
   }
 
-  /// Get the absolute position, with the anchor taken into consideration
-  Vector2 get absolutePosition => absoluteParentPosition + position;
-
-  /// Get the relative top left position regardless of the anchor and angle
-  Vector2 get topLeftPosition {
-    return anchor.toOtherAnchorPosition(
-      position,
-      Anchor.topLeft,
-      scaledSize,
-    );
+  /// The total transformation matrix for the component. This matrix combines
+  /// translation, rotation and scale transforms into a single entity. The
+  /// matrix is cached and gets recalculated only as necessary.
+  Matrix4 get transformMatrix => _transform.transformMatrix;
+
+  /// The position of this component's anchor on the screen.
+  NotifyingVector2 get position => _transform.position;
+  set position(Vector2 position) => _transform.position = position;
+
+  /// X position of this component's anchor on the screen.
+  double get x => _transform.x;
+  set x(double x) => _transform.x = x;
+
+  /// Y position of this component's anchor on the screen.
+  double get y => _transform.y;
+  set y(double y) => _transform.y = y;
+
+  /// Rotation angle (in radians) of the component. The component will be
+  /// rotated around its anchor point in the clockwise direction if the
+  /// angle is positive, or counterclockwise if the angle is negative.
+  double get angle => _transform.angle;
+  set angle(double a) => _transform.angle = a;
+
+  /// Rotation angle (in degrees) of the component. The component will be
+  /// rotated around its anchor point in the clockwise direction if the
+  /// angle is positive, or counterclockwise if the angle is negative.
+  double get angleDegrees => _transform.angleDegrees;
+  set angleDegrees(double a) => _transform.angleDegrees = a;
```

Comment:
I think we intentionally left this out to not have a mix of radians and degrees throughout the games that are made.
Do you rememember @flame-engine/flame-colabc

Reply:
You think there's potential for confusion? It's always easy to remove this property if need be.

Reply:
We did indeed decide to not introduce two types of angles. And we are using only radians throughout the whole engine.

Reply:
ok, I'll remove these methods then